geo: bump geos and twpayne/go-geom

otan opened the pull request:
* Bump GEOS to parse MultiPoint containing empty elements.
* Bump twpayne/go-geom to 1.3.6 to fix handling of EMPTY element
  MULTIPOLYGONs.
* Fix respective tests.

Resolves #53997
Resolves #53998

Release justification: low risk bug fixes to new functionality.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54033)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54033#-:-MGiipk-9DxhOp00xGIB:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geomfn/linestring.go, line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54033#-MGii2Fr8ZDQatCM5GpG:-MGii2Fr8ZDQatCM5GpH:b-u9010j) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f84d294c47e9731de2576ca5208280ec783ba5e/pkg/geo/geomfn/linestring.go#L38)):*
> ```Go
> 	for i := 0; i < mp.NumPoints(); i++ {
> 		p := mp.Point(i)
> 		// Empty points in multipoints double count the previous coordiante.
> ```

typo: coordinate

I didn't understand this logic. I had to look at the test case to understand. Can you add an example to this comment and clarify that this double counting or using (0, 0) is the conversion semantics (the comment seems to imply it is multipoint semantics).


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54033)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @sumeerbhola)

___
*[pkg/geo/geomfn/linestring.go, line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54033#-MGii2Fr8ZDQatCM5GpG:-MGisNOEDD-LfI7wx7pf:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f84d294c47e9731de2576ca5208280ec783ba5e/pkg/geo/geomfn/linestring.go#L38)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

typo: coordinate

I didn't understand this logic. I had to look at the test case to understand. Can you add an example to this comment and clarify that this double counting or using (0, 0) is the conversion semantics (the comment seems to imply it is multipoint semantics).
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola

otan commented:
bors r-

otan committed:
geo: bump geos and twpayne/go-geom

* Bump GEOS to parse MultiPoint containing empty elements.
* Bump twpayne/go-geom to 1.3.6 to fix handling of EMPTY element
  MULTIPOLYGONs.
* Fix respective tests.

Release justification: low risk bug fixes to new functionality.

Release note: None

otan commented:
bors r=sumeerbhola
